src/index.ts

```diff
@@ -1,16 +1,24 @@
 import * as dataTypes from './dataTypes';
-import { Map, TileLayer, GeoJSON } from 'leaflet';
+// TODO karma complains if import from 'geoJson'. Figure out more elegant solution.
+import * as geoJson from '../node_modules/@types/geojson/index';
+import * as leafletTypes from '../node_modules/@types/leaflet/index';
 
 export interface ILayerReference {
     [state: string]: dataTypes.ILeafletLayer;
 }
 
 export class DeclarativeLayers {
-    private map: Map;
+    private map: leafletTypes.Map;
     private layerReferences: ILayerReference = {}; // reference for removal from map
+    private suppliedLeafletReference: any;
 
-    constructor(targetMap: Map, layersMetadata?: dataTypes.ILayersMetadata) {
+    constructor(
+            suppliedLeafletReference: any,
```

Comment:
remove `any`?

Also, to keep code cleaner, you can do the following in TS

```ts
export class DeclarativeLayers {
  private suppliedLeafletReference: any,
  private map: leafletTypes.map
  constructor(
    suppliedLeafletReference: any,
    map: leafletTypes.map
  ) {
    this.suppliedLeafletReference = suppliedLeafletReference;
    this.map = map;
  }
}

// equivalent to
export class DeclarativeLayers {
  constructor(
    private suppliedLeafletReference: any,
    private map: leafletTypes.map
  ) {}
}

```

Reply:
Nice!